client/tests/webdriver/specs/customFields.spec.js

```diff
@@ -0,0 +1,274 @@
+
+    it("Should persist previous valid data when toggling field's visibility", () => {
+      CreatePerson.numberCustomField.setValue(validNumberInput)
+      // make default invisible fields invisible again, amber color does that
+      CreatePerson.amberButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist({ reverse: true })
+      // make visible
+      CreatePerson.greenButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist()
+
+      expect(CreatePerson.numberCustomField.getValue()).be.equal(
+        validNumberInput
+      )
+    })
+
+    it("Should not persist previous invalid data when toggling field's visibility", () => {
+      CreatePerson.numberCustomField.setValue(inValidNumberInput)
+      // Actually see the validation warning
+      CreatePerson.numberCustomFieldHelpText.waitForExist()
+      // make invisible
+      CreatePerson.amberButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist({ reverse: true })
+      // make visible
+      CreatePerson.greenButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist()
+
+      expect(CreatePerson.numberCustomField.getValue()).not.be.equal(
+        inValidNumberInput
+      )
+      expect(CreatePerson.numberCustomField.getValue()).to.equal("")
+    })
+
+    it("Should validate visible field", () => {
+      CreatePerson.numberCustomField.setValue(inValidNumberInput)
+      CreatePerson.numberCustomFieldHelpText.waitForExist()
+    })
+
+    it("Should not validate invisible field so we can submit the form", () => {
+      // make invisible
+      CreatePerson.amberButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist({ reverse: true })
+      CreatePerson.submitForm()
+      CreatePerson.waitForAlertSuccessToLoad()
+    })
+  })
+  // ------------------------------ TASK CUSTOM FIELDS -----------------------------------------
+  describe("For task's custom fields", () => {
+    it("Should open create new task page", () => {
+      CreateTask.openAsAdmin()
+      CreateTask.form.waitForExist()
+      CreateTask.form.waitForDisplayed()
+      // Fill other required fields so that we can test custom field validation
+      CreateTask.shortName.setValue(REQUIRED_TASK_FIELDS.shortName)
+    })
+
+    it("Should be able to see assessment fields", () => {
+      CreateTask.addObjectButton.click()
+      CreateTask.objectFields.forEach(field => {
+        field.waitForExist()
+      })
+    })
+
+    it("Should warn invalid json for questions", () => {
+      CreateTask.questionsField.setValue("invalidJsonTest")
+      // normally there is already a help block, we need other warning text
+      CreateTask.questionsFieldWarningText.waitForExist()
+    })
+
+    it("Should not warn valid json for questions", () => {
+      CreateTask.questionsField.setValue("{}")
+      // all help texts should be removed with valid json
+      CreateTask.questionsFieldHelpText.waitForExist({ reverse: true })
```

Comment:
I think this only works because the selector defined for questionsFieldHelpText is not correct.